abi/src/main/java/org/web3j/abi/EventEncoder.java

```diff
@@ -20,21 +20,21 @@
     private EventEncoder() { }
 
     public static String encode(Event function) {
-        List<TypeReference<Type>> indexedParameters = function.getIndexedParameters();
         List<TypeReference<Type>> nonIndexedParameters = function.getNonIndexedParameters();
+        List<TypeReference<Type>> indexedParameters = function.getIndexedParameters();
 
         String methodSignature = buildMethodSignature(function.getName(),
-                indexedParameters, nonIndexedParameters);
+                nonIndexedParameters, indexedParameters);
 
         return buildEventSignature(methodSignature);
     }
 
     static <T extends Type> String buildMethodSignature(
```

Comment:
A unit test should accompany any changes.

Reply:
will do this asap